packages/node/src/integrations/tracing/firebase/otel/patches/firestore.ts

```diff
@@ -0,0 +1,299 @@
+import type { FirebaseApp, FirebaseOptions } from '@firebase/app';
+import type {
+  CollectionReference,
+  DocumentData,
+  DocumentReference,
+  FirestoreSettings,
+  PartialWithFieldValue,
+  QuerySnapshot,
+  SetOptions,
+  WithFieldValue,
+  addDoc,
+  deleteDoc,
+  getDocs,
+  setDoc,
+} from '@firebase/firestore';
```

Comment:
Before we merge, let's make sure that this doesn't end up in the `.d.ts` in any way.